extensions/rvdatabinding/src/main/java/com/google/android/agera/rvdatabinding/DataBindingRepositoryPresenterCompiler.java

```diff
@@ -137,11 +151,44 @@ public Object stableIdForItem(@NonNull final Function stableIdForItem) {
     @Override
     public void bind(@NonNull final Object item, @NonNull final View view) {
       final ViewDataBinding viewDataBinding = DataBindingUtil.bind(view);
-      viewDataBinding.setVariable(itemId.apply(item), item);
+      final Integer itemVariable = itemId.apply(item);
+      viewDataBinding.setVariable(itemVariable, item);
+      view.setTag(R.id.dataBindingRepositoryItem, itemVariable);
       for (final Pair<Integer, Object> handler : handlers) {
         viewDataBinding.setVariable(handler.first, handler.second);
       }
       viewDataBinding.executePendingBindings();
     }
   }
+
+  private static final class ViewRecycler implements Receiver<View> {
+    @RecycleConfig
+    private final int recycleConfig;
+    @NonNull
+    private List<Pair<Integer, Object>> handlers;
+
+    ViewRecycler(
+        @RecycleConfig final int recycleConfig,
+        @NonNull final List<Pair<Integer, Object>> handlers) {
+      this.recycleConfig = recycleConfig;
+      this.handlers = handlers;
+    }
+
+    @Override
+    public void accept(@NonNull final View view) {
+      if (recycleConfig == CLEAR_ITEM || recycleConfig == CLEAR_ALL) {
+        final ViewDataBinding viewDataBinding = DataBindingUtil.bind(view);
+        final Object tag = view.getTag(R.id.dataBindingRepositoryItem);
+        if (tag instanceof Integer) {
```

Comment:
it's really just a fancy tag != null, if we set it, it'll always be an Integer. wouldn't want to alloc a new wrapper for each bind either, I'm hoping this is safe enough?